sdks/python/apache_beam/testing/benchmarks/nexmark/nexmark_launcher.py

```diff
@@ -0,0 +1,245 @@
+                        help='Path to the data file containing nexmark events.')
+
+    self.args, self.pipeline_args = parser.parse_known_args()
+    logging.basicConfig(level=getattr(logging, self.args.loglevel, None),
+                        format='(%(threadName)-10s) %(message)s')
+
+    self.pipeline_options = PipelineOptions(self.pipeline_args)
+    logging.debug('args, pipeline_args: %s, %s', self.args, self.pipeline_args)
+
+    # Usage with Dataflow requires a project to be supplied.
+    self.project = self.pipeline_options.view_as(GoogleCloudOptions).project
+    if self.project is None:
+      parser.print_usage()
+      print(sys.argv[0] + ': error: argument --project is required')
+      sys.exit(1)
+
+    # Pub/Sub is currently available for use only in streaming pipelines.
+    self.streaming = self.pipeline_options.view_as(StandardOptions).streaming
+    if self.streaming is None:
+      parser.print_usage()
+      print(sys.argv[0] + ': error: argument --streaming is required')
+      sys.exit(1)
+
+    # wait_until_finish ensures that the streaming job is canceled.
+    self.wait_until_finish_duration = (
+        self.pipeline_options.view_as(TestOptions).wait_until_finish_duration
+    )
+    if self.wait_until_finish_duration is None:
+      parser.print_usage()
+      print(sys.argv[0] + ': error: argument --wait_until_finish_duration is required') # pylint: disable=line-too-long
+      sys.exit(1)
+
+    # We use the save_main_session option because one or more DoFn's in this
+    # workflow rely on global context (e.g., a module imported at module level).
+    self.pipeline_options.view_as(SetupOptions).save_main_session = True
+
+  def generate_events(self):
+    publish_client = pubsub.Client(project=self.project)
+    topic = publish_client.topic(self.topic_name)
+    if topic.exists():
+      topic.delete()
+    topic.create()
+    sub = topic.subscription(self.subscription_name)
+    if sub.exists():
+      sub.delete()
+    sub.create()
+
+    logging.info('Generating auction events to topic %s', topic.name)
+
+    if self.args.input.startswith('gs://'):
+      from apache_beam.io.gcp.gcsfilesystem import GCSFileSystem
+      fs = GCSFileSystem(self.pipeline_options)
+      with fs.open(self.args.input) as infile:
+        for line in infile:
+          topic.publish(line)
+    else:
+      with open(self.args.input) as infile:
+        for line in infile:
+          topic.publish(line)
+
+    logging.info('Finished event generation.')
+
+    # Read from PubSub into a PCollection.
+    if self.args.subscription_name:
+      raw_events = self.pipeline | 'ReadPubSub' >> beam.io.ReadFromPubSub(
+          subscription=sub.full_name)
+    else:
+      raw_events = self.pipeline | 'ReadPubSub' >> beam.io.ReadFromPubSub(
+          topic=topic.full_name)
+
+    return raw_events
+
+  def run_query(self, query, query_errors):
+    try:
+      self.pipeline = beam.Pipeline(options=self.pipeline_options)
```

Comment:
pipeline options are not reusable across pipelines. At least for DataflowRunner this will work for the first it is executed (i.e. first query in the list) and may have issues for the successive runs.

Reply:
Ping on this. Please fix in the next PR, because it will become an issue as you add more queries.